crates/compute/src/cpu/layer.rs

```diff
@@ -7,14 +7,77 @@ use binius_field::{
 };
 
 use super::{memory::CpuMemory, tower_macro::each_tower_subfield};
-use crate::layer::{ComputeLayer, Error, FSlice, FSliceMut};
+use crate::{
+	layer::{ComputeLayer, Error, FSlice, FSliceMut},
+	memory::FSliceWithTowerLevel,
+};
 
 #[derive(Debug)]
 pub struct CpuExecutor;
 
 #[derive(Debug, Default)]
 pub struct CpuLayer<F: TowerFamily>(PhantomData<F>);
 
+impl<F: TowerFamily> CpuLayer<F> {
+	pub fn new() -> Self {
+		Self(PhantomData)
+	}
+}
+
+/// evals is treated as a matrix with `1 << log_query_size` rows and each column is dot-producted
+/// with the corresponding query element. The result is written to the `output` slice of packed values.
+fn compute_left_fold<EvalType: TowerField, T: TowerFamily>(
+	evals_as_b128: Vec<T::B128>,
+	log_evals_size: usize,
```

Comment:
Please add a comment, or indicate otherwise, that evals can in fact be anything from 1b to 128b. Otherwise, the function signature looks like evals are exactly 128b elements.